service-base/src/main/java/org/eclipse/hono/service/credentials/CredentialsAmqpEndpoint.java

```diff
@@ -32,7 +32,7 @@
  * It receives AMQP 1.0 messages representing requests and sends them to an address on the vertx
  * event bus for processing. The outcome is then returned to the peer in a response message.
  */
-public final class CredentialsAmqpEndpoint extends RequestResponseEndpoint<ServiceConfigProperties> {
+public class CredentialsAmqpEndpoint extends RequestResponseEndpoint<ServiceConfigProperties> {
```

Comment:
Please make the `getName()` method final, too.

Reply:
No. It was intentional so that overriding classes can give a custom name.

Reply:
That is about obvious - can you explain a little why that makes sense?
IMHO if you customize the, let's say, credentials endpoint - why do you need another name for that? Would you have more than one credentials endpoint (the original one AND the customized one?).
This is currently not really clear to me ;-)

Reply:
We could name it differently, say with a version in it and we know when we instantiate, which we instantiated.

Reply:
I kept it quite similar to https://github.com/eclipse/hono/blob/master/service-base/src/main/java/org/eclipse/hono/service/amqp/AmqpServiceBase.java#L74

Reply:
The `getName()` is currently mainly used for logging output of the endpoint, BUT:
not only. Which might be a piece to refactor, but since it is currently not prepared for that, I do not see the importance to let this be customized.
With the current code, there might be problems if you customize the name, but do not customize the corresponding targetAddresses the same way - so I currently stick to keeping the `getName()` method `final`.

Reply:
Done